### PR TITLE
[roslib] Args for cache the result of path finding in catkin.find_in_workspaces

### PR DESCRIPTION
**Pull request by wkentaro**
Closes #74

For https://github.com/ros/ros_comm/pull/676

### CODE REVIEW

File: core/roslib/src/roslib/packages.py [hunk 1]
```diff
@@ -384,20 +384,24 @@ def list_pkgs_by_path(path, packages=None, cache=None, env=None):
...
-def find_node(pkg, node_type, rospack=None):
+def find_node(pkg, node_type, rospack=None, source_path_to_packages=None):
     """
     Warning: unstable API due to catkin.
 
     Locate the executable that implements the node
     
     :param node_type: type of node, ``str``
+    :param source_path_to_packages: the dictionary is populated with mappings
+        from source paths to packages.
```
Comment (dirk-thomas) [inline core/roslib/src/roslib/packages.py]: Please use the same documentation as in https://github.com/ros/catkin/blob/3b8a3b9607adff98ed32cfd3b6d6ea4aed647424/python/catkin/find_in_workspaces.py#L106

Same below.

Comment (wkentaro) [inline core/roslib/src/roslib/packages.py]: Ok. I amended the commit.

**Comment by wkentaro** [comment]
I updated the commit, so please review again.

**Comment by dirk-thomas** [comment]
Thank you.

**Comment by dirk-thomas** [comment]
Thank you. I have cherry-picked the patch to the _jade-devel_ branch (a59b53a282b5df112ee098e6cbc636767176e9e0). But it will also be backported before the next Indigo patch release.

**Comment by dirk-thomas** [comment]
I have reverted the changes from this PR in favor of #95.

**Pull request closed without merging**